examples/piecestore/rpc/server/main.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"path"
+	"regexp"
+
+	_ "github.com/mattn/go-sqlite3"
+	"google.golang.org/grpc"
+
+	"storj.io/storj/pkg/piecestore/rpc/server"
+	"storj.io/storj/pkg/piecestore/rpc/server/ttl"
+	pb "storj.io/storj/protos/piecestore"
+)
+
+func main() {
+	port := "7777"
+
+	if len(os.Args) > 1 && os.Args[1] != "" {
+		if matched, _ := regexp.MatchString(`^\d{2,6}$`, os.Args[1]); matched == true {
+			port = os.Args[1]
+		}
+	}
+
+	dataFolder := "./"
+
+	if len(os.Args) > 2 && os.Args[2] != "" {
+		dataFolder = os.Args[2]
+	}
+
+	fileInfo, err := os.Stat(dataFolder)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	if fileInfo.IsDir() != true {
+		log.Fatalf("dataFolder %s is not a directory", dataFolder)
+	}
+
+	dataDir := path.Join(dataFolder, "piece-store-data/", port)
+
+	ttlDB, err := ttl.NewTTL("ttl-data.db")
+	if err != nil {
+		log.Fatalf("failed to open DB")
+	}
+
+	// create a listener on TCP port
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%s", port))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+
```

Comment:
defer lis.Close()

not important enough to yada yada you get it

Reply:
👍